components/kyma-environment-broker/internal/storage/dbsession/read.go

```diff
@@ -1,18 +1,42 @@
 package dbsession
 
 import (
-	dbr "github.com/gocraft/dbr"
+	"fmt"
+
 	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
 	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage/dberr"
 	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage/dbsession/dbmodel"
 	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage/postsql"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage/predicate"
+
+	"github.com/gocraft/dbr"
 	"github.com/pivotal-cf/brokerapi/v7/domain"
 )
 
 type readSession struct {
 	session *dbr.Session
 }
 
+func (r readSession) FindAllInstancesJoinedWithProvisionOperation(prct ...predicate.Predicate) ([]internal.InstanceWithOperation, dberr.Error) {
```

Comment:
it seems it will also join with deprovisioning operation, is that ok?

Reply:
yes, check the testdata result there is also a test for that :)